### PR TITLE
Add Redis metrics and Spans data to New Relic exports

### PR DESCRIPTION
**Pull request by philkuz**
Signed-off-by: Phillip Kuznetsov <pkuznetsov@pixielabs.ai>

### CODE REVIEW

File: plugins/new-relic/retention.yaml [hunk 1]
```diff
@@ -808,4 +808,189 @@ presetScripts:
...
+        resource={
+          'service.name': otel_df.source_service,
+          'service.instance.id': otel_df.source_pod,
+          'k8s.namespace.name': otel_df.source_namespace,
+          'redis.service.name': otel_df.destination_service,
+          'redis.pod.name': otel_df.destination_pod,
+          'redis.namespace.name': otel_df.destination_namespace,
+          'redis.req_cmd': otel_df.req_cmd,
+          'k8s.cluster.name': otel_df.cluster_name,
+          'px.cluster.id': otel_df.cluster_id,
```
Comment (htroisi) [inline plugins/new-relic/retention.yaml:971]: Should this be `pixie.cluster.id` for readability?

Comment (philkuz) [reply plugins/new-relic/retention.yaml:971]: we use px.cluster.id elsewhere already

Comment (htroisi) [reply plugins/new-relic/retention.yaml:971]: Let's standardize on pixie.cluster.id: https://github.com/pixie-io/pixie-plugin/pull/29

File: plugins/new-relic/retention.yaml [hunk 2]
```diff
@@ -808,4 +808,189 @@ presetScripts:
...
+          latency_min=('latency', px.min),
+          latency_max=('latency', px.max),
+          latency_sum=('latency', px.sum),
+          time_=('time_', px.max),
+          req_bytes=('req_bytes', px.sum),
+          resp_bytes=('resp_bytes', px.sum),
+      )
+
+      df.destination_namespace = px.pod_name_to_namespace(df.destination_pod)
+      df.source_namespace = px.pod_name_to_namespace(df.source_pod)
```
Comment (htroisi) [inline plugins/new-relic/retention.yaml:954]: Need to strip the namespace from Pixie's pod & service names:
```
df.source_namespace = px.pod_name_to_namespace(df.source_pod)
ns_prefix = df.source_namespace + '/'
df.source_pod = px.strip_prefix(ns_prefix, df.source_pod)
df.source_service = px.strip_prefix(ns_prefix, df.source_service)

df.destination_namespace = px.pod_name_to_namespace(df.destination_pod)
ns_prefix = df.destination_namespace + '/'
df.destination_pod = px.strip_prefix(ns_prefix, df.destination_pod)
df.destination_service = px.strip_prefix(ns_prefix, df.destination_service)
```

Comment (philkuz) [reply plugins/new-relic/retention.yaml:954]: this doesn't work unfortunately

Comment (philkuz) [reply plugins/new-relic/retention.yaml:954]:
```
def remove_ns_prefix(column):
    return px.replace('[a-z0-9\-]*/', column, '')

df.pod = remove_ns_prefix(df.pod)
```

File: plugins/new-relic/retention.yaml (hunk 1 above)
Comment (philkuz) [reply plugins/new-relic/retention.yaml:988]: note: we switched back to px.cluster.id

File: plugins/new-relic/retention.yaml [hunk 3]
```diff
@@ -808,4 +808,206 @@ presetScripts:
...
+          'service.name': otel_df.source_service,
+          'service.instance.id': otel_df.source_pod,
+          'k8s.namespace.name': otel_df.source_namespace,
+          'redis.service.name': otel_df.destination_service,
+          'redis.pod.name': otel_df.destination_pod,
+          'redis.namespace.name': otel_df.destination_namespace,
+          'redis.req_cmd': otel_df.req_cmd,
+          'k8s.cluster.name': otel_df.cluster_name,
+          'px.cluster.id': otel_df.cluster_id,
+          'instrumentation.provider': otel_df.pixie,
```
Comment (htroisi) [inline plugins/new-relic/retention.yaml:887]: Should we add `df.db_system = 'redis'` to both scripts?

Comment (philkuz) [reply plugins/new-relic/retention.yaml:887]: db_system might be a specific catch all for the SQL nerdlet. redis wouldn't work properly there.

**Review by aimichelle — APPROVED** [review]

**Pull request merged**